src/harness/testcases/heartbeat_testcase.py

```diff
@@ -824,3 +824,123 @@ def test_WINNF_FT_S_HBT_18(self):
                            grant_expire_times[response_num])
     self.assertEqual(response[2]['response']['responseCode'], 103)
     # No need to check transmitExpireTime because this is the first heartbeat
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_19(self):
+    """Heartbeat Request from CBSD in Granted or Authorized state 
+    (immediately after CBSD moves into Granted State or following
+     a Heartbeat Response) requires CBSD to de-register. 
+
+    The response should be FAIL, code 105."""
+
+    # Register the device
+    device_a = json.load(open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+
+    request = {'registrationRequest': [device_a]}
+    response = self._sas.Registration(request)['registrationResponse'][0]
+    # Check registration response
+    self.assertEqual(response['response']['responseCode'], 0)
+    cbsd_id = response['cbsdId']
+    del request, response
+
+    # Request grant
+    grant_0 = json.load(open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_id
+    request = {'grantRequest': [grant_0]}
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse'][0]
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertTrue(response['grantId'])
+    self.assertEqual(response['response']['responseCode'], 0)
+    grant_id = response['grantId']
+    del request, response
+
+    # Inject Device into Blacklist
+    self._sas_admin.BlacklistByFccId({'fccId': device_a['fccId']})
+
+    # Request Heartbeat
+    request = {
+        'heartbeatRequest': [{
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': 'GRANTED'
+        }]
+    }
+    response = self._sas.Heartbeat(request)['heartbeatResponse'][0]
+    # Check the first heartbeat response
+    self.assertEqual(response['cbsdId'], cbsd_id)
+    self.assertEqual(response['grantId'], grant_id)
+    self.assertEqual(response['response']['responseCode'], 105)
+    del request, response
```

Comment:
Does the checking of cbsdId and grantId optional here? Test spec mentions to validate the cbsdId and grantId 
![image](https://cloud.githubusercontent.com/assets/23145897/25584592/a9beee50-2e64-11e7-96cf-e08202ab6fce.png)